src/plugins/orangehrmDashboardPlugin/Service/ChartService.php

```diff
@@ -73,39 +73,28 @@ static function (SubunitCountPair $x, SubunitCountPair $y) {
         );
     }
 
-    /**
-     * @param array $locationEmployeeCounts
-     * @return int
-     */
-    public function getLocationUnassignedEmployeeCount(array $locationEmployeeCounts): int
-    {
-        $totalActiveEmployee = $this->getChartDao()->getTotalActiveEmployeeCount();
-
-        $assignedEmployeeCount = 0;
-        foreach ($locationEmployeeCounts as $locationEmployeeCount) {
-            $assignedEmployeeCount += $locationEmployeeCount->getEmployeeCount();
-        }
-        return $totalActiveEmployee - $assignedEmployeeCount;
-    }
-
     /**
      * @param int $limit
      * @return EmployeeDistributionByLocation
      */
     public function getEmployeeDistributionByLocation(int $limit=8): EmployeeDistributionByLocation
     {
         $locationEmployeeCount = $this->getChartDao()->getEmployeeDistributionByLocation();
-        $unassignedEmployeeCount =  $this->getLocationUnassignedEmployeeCount($locationEmployeeCount);
 
-        $otherArray = [];
-        if (count($locationEmployeeCount) > $limit) {
-            $otherArray = array_slice($locationEmployeeCount, $limit);
-            $locationEmployeeCount = array_slice($locationEmployeeCount, 0, $limit);
+        $otherCount = 0;
+        $assignedEmployeeCount = 0;
+        foreach ($locationEmployeeCount as $key => $employeeCount) {
+            $assignedEmployeeCount += $employeeCount->getEmployeeCount();
+            if ($key >= $limit) {
+                $otherCount += $employeeCount->getEmployeeCount();
+            }
         }
 
-        $otherCount = 0;
-        foreach ($otherArray as $locationCountPair) {
-            $otherCount += $locationCountPair->getEmployeeCount();
+        $totalActiveEmployee = $this->getChartDao()->getTotalActiveEmployeeCount();
+        $unassignedEmployeeCount = $totalActiveEmployee - $assignedEmployeeCount;
+
+        if (count($locationEmployeeCount) > $limit) {
```

Comment:
```suggestion
        if (count($locationEmployeeCount) > $limit + 1) {
```
refactor carefully and check unit tests. if you can't add multiple service unit tests. check code coverage after these bug fixes